identity/aziot-identityd/src/lib.rs

```diff
@@ -414,99 +457,47 @@ impl Api {
                 scope_id,
                 attestation,
             } => {
+                let dps_client = aziot_dps_client_async::Client::new(
+                    &global_endpoint,
+                    &scope_id,
+                    self.key_client.clone(),
+                    self.key_engine.clone(),
+                    self.cert_client.clone(),
+                );
+
                 let device = match attestation {
+                    settings::DpsAttestationMethod::TPM { registration_id } => {
```

Comment:
What's the point of adding this in this PR instead of your tpmd one?

Reply:
I guess it goes back to the age old debate between "one big PR vs. multiple smaller PRs". Having one big PR for the TPM work makes sense, since reviewing/testing it requires having the full E2E flow implemented. This PR doesn't actually add any new functionality, and just refactors existing code, so I thought I'd break it out into it's own smaller PR.

Reply:
If we merge this today and then tomorrow we decide "Let's not merge Daniel's PR because we've decided to stop supporting TPM provisioning", what will have been the point of adding this code?

Adding any code related to TPM provisioning is unrelated to a PR that refactors existing code. In fact it is the opposite of what a refactoring PR should do.

Reply:
Oh, sorry, I totally didn't realize that this was a comment specifically about the TPM bit.

Yeah, my bad, I don't know what I was thinking including this in the PR (probably lazyness). I'll got ahead and remove it.